src/loaders/stm_load.c

```diff
@@ -270,7 +270,8 @@ static int stm_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	D_(D_INFO "Stored samples: %d", mod->smp);
 
 	for (i = 0; i < mod->ins; i++) {
-		if (mod->xxs[i].len > 1) {
+		if (sfh.ins[i].volume && sfh.ins[i].length) {
```

Comment:
Volume check exists in original Scream Tracker 2 loader, i have disassemled it. Problem is not in garbage samples (in fact there is no "garbage samples", ST2 does not load sample if it's volume and length equals zero), but in that the sfh.ins[i].rsvd1 << 4 is actually sample offset. Try to load wonder.stm from my attachement and you will be surprised.